### PR TITLE
Directly reference DBNull

### PR DESCRIPTION
**Pull request by weshaggard**
Clean up some old reflection references to DBNull and statically
reference it now.

cc @sepidehMS @krwq @shmao

### CODE REVIEW

**Comment by weshaggard** [comment]
>LGTM, @weshaggard do we still need [SecuritySafeCritical]?

Nope I will remove those as well.

**Comment by weshaggard** [comment]
Innerloop PortableLinux Release x64 Build and Test
Innerloop Windows_NT Debug x86 Build and Test

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Globals.cs [hunk 1]
```diff
@@ -778,35 +778,20 @@ internal static Type TypeOfXmlNodeArray
             }
         }
 
-        private static bool s_shouldGetDBNullType = true;
-
-        private static Type s_typeOfDBNull;
+        private static Type s_typeOfDBNull = typeof(DBNull);
```
Comment (stephentoub) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Globals.cs]: Do we even need the static, or could the usage sites just use type of directly?

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Globals.cs [hunk 2]
```diff
@@ -778,35 +778,20 @@ internal static Type TypeOfXmlNodeArray
...
-                if (s_valueOfDBNull == null && TypeOfDBNull != null)
-                {
-                    var fieldInfo = TypeOfDBNull.GetField("Value");
-                    if (fieldInfo != null)
-                        s_valueOfDBNull = fieldInfo.GetValue(null);
-                }
-
-                return s_valueOfDBNull;
+                return DBNull.Value;
             }
```
Comment (stephentoub) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Globals.cs]: Could this property just be replaced at its call sites?

File: src/System.Private.Xml/src/System/Xml/Serialization/Globals.cs [hunk 3]
```diff
@@ -16,29 +16,20 @@ namespace System.Xml.Serialization
...
-        private static object s_valueOfDBNull;
         internal static object ValueOfDBNull
         {
-            [SecuritySafeCritical]
             get
             {
-                if (s_valueOfDBNull == null && TypeOfDBNull != null)
-                    s_valueOfDBNull = TypeOfDBNull.GetField("Value").GetValue(null);
-                return s_valueOfDBNull;
+                return DBNull.Value;
```
Comment (stephentoub) [inline src/System.Private.Xml/src/System/Xml/Serialization/Globals.cs]: Same comments/questions

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Globals.cs (hunk 1 above)
Comment (weshaggard) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Globals.cs]: We could probably update the sites directly.  I will see how many places that is.

**Comment by shmao** [comment]
LGTM.

**Comment by weshaggard** [comment]
I updated the code to make it match what we have in desktop. Which means no Globals.TypeOfDBNull in the xml serialization but they do still have one in the datacontract serialization so I left that one there.

**Pull request merged**